odoo/addons/test_read_group/tests/test_fill_temporal.py

```diff
@@ -0,0 +1,746 @@
+                    {'__domain': ['&',
+                                  ('date', '>=', '1915-11-01'),
+                                  ('date', '<', '1915-12-01')],
+                     'date': 'November 1915',
+                     'date_count': 0,
+                     'value': False},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1915-12-01'),
+                                  ('date', '<', '1916-01-01')],
+                     'date': 'December 1915',
+                     'date_count': 0,
+                     'value': False},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-01-01'),
+                                  ('date', '<', '1916-02-01')],
+                     'date': 'January 1916',
+                     'date_count': 1,
+                     'value': 5}]
+
+        # Time Zone                      UTC     UTC DST
+        tzs = ["America/Anchorage",  # −09:00    −08:00
+               "Europe/Brussels",    # +01:00    +02:00
+               "Pacific/Kwajalein"]  # +12:00    +12:00
+
+        for tz in tzs:
+            groups = self.Model.with_context(tz=tz, fill_temporal=True).read_group([],
+                                                                                   fields=['date',
+                                                                                           'value'],
+                                                                                   groupby=['date'])
+            self.assertEqual(groups, expected)
+
+    def test_only_with_undefined(self):
+        """We should have the same result when fill_temporal is set or not."""
+        self.Model.create({'date': False, 'value': 13})
+        self.Model.create({'date': False, 'value': 11})
+        self.Model.create({'date': False, 'value': 17})
+
+        expected = [{'__domain': [('date', '=', False)],
+                     'date_count': 3,
+                     'value': 41,
+                     'date': False}]
+
+        groups = self.Model.with_context(fill_temporal=False).read_group([],
+                                                                         fields=['date',
+                                                                                 'value'],
+                                                                         groupby=['date'])
+        self.assertEqual(groups, expected)
+
+        groups = self.Model.with_context(fill_temporal=True).read_group([],
+                                                                        fields=['date',
+                                                                                'value'],
+                                                                        groupby=['date'])
+        self.assertEqual(groups, expected)
+
+    def test_date_range_and_undefined(self):
+        """Similar test than test_only_with_undefined but with setted dates in more."""
```

Comment:
The docstring is a bit difficult to parse. Leave "undefined" to JS, and call it a **null value** instead:
```
Test data with null and non-null dates.
```